### PR TITLE
fix(Popup): Do not propagate keydown events outside Popup's content

### PR DESCRIPTION
**Pull request by sophieH29**
Closes https://github.com/stardust-ui/react/issues/978

No need to propagate keydown events outside Popup, allow only keyboard actions to execute

### CODE REVIEW

File: packages/react/src/components/Popup/Popup.tsx [hunk 1]
```diff
@@ -439,8 +438,12 @@ export default class Popup extends AutoControlledComponent<ReactProps<PopupProps
     const popupWrapperAttributes = {
       ...(rtl && { dir: 'rtl' }),
       ...accessibility.attributes.popup,
-      ...accessibility.keyHandlers.popup,
-
+      onKeyDown: (e: KeyboardEvent) => {
+        // No need to propagate keydown events outside Popup
+        // allow only keyboard actions to execute
+        _.invoke(accessibility.keyHandlers.popup, 'onKeyDown', e)
+        e.stopPropagation()
```
Comment (bmdalex) [inline packages/react/src/components/Popup/Popup.tsx:445]: is this testable? would be great to have a UT since we had problems with Popup

**Review by bmdalex — APPROVED** [review]
👍

File: packages/react/src/components/Popup/Popup.tsx (hunk 1 above)
Comment (layershifter) [reply packages/react/src/components/Popup/Popup.tsx:445]: Yep, you can do:
```
const stopPropagation = jest.fn()

wrapper.find('CONTENT').simulate('keyDown', { stopPropagation })
expect(stopPropagation).toHaveBeenCalledTimes(1)
```

File: packages/react/src/components/Popup/Popup.tsx [hunk 2]
```diff
@@ -439,8 +438,12 @@ export default class Popup extends AutoControlledComponent<ReactProps<PopupProps
     const popupWrapperAttributes = {
       ...(rtl && { dir: 'rtl' }),
       ...accessibility.attributes.popup,
-      ...accessibility.keyHandlers.popup,
-
+      onKeyDown: (e: KeyboardEvent) => {
```
Comment (layershifter) [inline packages/react/src/components/Popup/Popup.tsx]:
```suggestion
      onKeyDown: (e: React.KeyboardEvent) => {
```

**Review by layershifter — APPROVED** [review]

File: packages/react/src/components/Popup/Popup.tsx [hunk 3]
```diff
@@ -439,8 +438,12 @@ export default class Popup extends AutoControlledComponent<ReactProps<PopupProps
     const popupWrapperAttributes = {
       ...(rtl && { dir: 'rtl' }),
       ...accessibility.attributes.popup,
-      ...accessibility.keyHandlers.popup,
-
+      onKeyDown: (e: React.KeyboardEvent) => {
```
Comment (mnajdova) [inline packages/react/src/components/Popup/Popup.tsx:441]: I know it is too late now, but I was just wondering whether our suggestion for the users would have been override the `onKeyDown` for the Popup and stop propagating if you want? This seem too much specific to the client's needs that we had at that moment.. @layershifter @sophieH29

**Pull request merged**